pkg/restore/restore.go

```diff
@@ -696,7 +772,21 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 
 			// Check if the PV exists in the cluster before attempting to create
 			// a volume from the snapshot, in order to avoid orphaned volumes (GH #609)
-			_, err := resourceClient.Get(name, metav1.GetOptions{})
+			pv, err := ctx.pvClient.Get(name, metav1.GetOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				addToResult(&errs, namespace, fmt.Errorf("error checking existence for PV %s: %v", name, err))
+			}
+
+			if isPVReleasedOrBound(pv) {
```

Comment:
I don't think `isPVReleasedOrBound` is quite right.  I think we want to wait if the PV has a phase of Released (or Failed, since sometimes they go to this phase and are subsequently deleted) *and* has a reclaim policy of Delete.  Does that sound right?  Is there something I'm overlooking?

Reply:
I think that makes sense. Would it make more sense to check if the PV has a reclaim policy of Delete and a phase that's not Terminating?

I included Bound because, depending on when the delete is issued, the PV may not have entered the Released state yet.